thruster/examples/route_params.rs

```diff
@@ -0,0 +1,39 @@
+use hyper::Body;
+use log::info;
+use thruster::{
+    context::{
+        basic_hyper_context::{generate_context, BasicHyperContext as Ctx, HyperRequest},
+        context_ext::ContextExt,
+    },
+    errors::{ErrorSet, ThrusterError},
+    hyper_server::HyperServer,
+    m, middleware_fn, App, Context, MiddlewareNext, MiddlewareResult, ThrusterServer,
+};
+
+#[middleware_fn]
+async fn echo_route_params(mut context: Ctx, _next: MiddlewareNext<Ctx>) -> MiddlewareResult<Ctx> {
+    context.set("Content-Type", "text/plain");
+    context.body = Body::from(format!(
+        "Hello, {}",
+        context
+            .get_params()
```

Comment:
Maybe the "get_" feels weird when compared to "query_params". I think "params" (so `context.params().get("name")`) feels more natural.

Other options for names:

- `parameters`: full name, feels kinda long?
- `path_params`: to differentiate with `query_params`, also is more specific about "where" it comes.
- `path_parameters`: Is the "official name" in https://swagger.io/specification/

Reply:
mm, yes, that's totally fair -- there's nothing else it _could_ mean, right? The `get_` prefix only makes sense when there's a setting function as well!